src/VNC.cpp

```diff
@@ -1166,7 +1336,7 @@ bool arduinoVNC::_handle_rre_encoded_message(rfbFramebufferUpdateRectHeader rect
             return false;
         display->draw_rect(
         Swap16IfLE(rect[0]) + rectheader.r.x,
-        Swap16IfLE(rect[1]) + rectheader.r.y, Swap16IfLE(rect[2]), Swap16IfLE(rect[3]), colour);
+        Swap16IfLE(rect[1]) + rectheader.r.y, Swap16IfLE(rect[2]), Swap16IfLE(rect[3]), Swap16IfLE(colour));

```

Comment:
is it not less CPU intensive to get the colour directly with the correct endianes via the opt.client.bigendian?

Reply:
Maybe a little bit, but making the endiannness dependend on the enconding seems to be way to confusing compared to a bit of byte swapping.